206/step2_3_recursion.py

```diff
@@ -0,0 +1,15 @@
+# Definition for singly-linked list.
+# class ListNode:
+#     def __init__(self, val=0, next=None):
+#         self.val = val
+#         self.next = next
+class Solution:
+    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
+        def reverse_from_node(node):
+            if node is None or node.next is None:
+                return node
+            reversed_head = reverse_from_node(node.next)
+            node.next.next = node
```

Comment:
node.next.next = node
という繋ぎ変えは何が起きているかぱっと見では理解しづらい可能性もあるので、冗長になってもnode.nextに変数名をつけて分かりやすくしてもよいかもしれません

```suggestion
            reversed_tail = node.next
            reversed_tail.next = node
```

Reply:
ご指摘ありがとうございます。
おっしゃるように、そのような記載の方が、後から見返したときに迷子にならなそうです。